Makefile

```diff
@@ -1,14 +1,43 @@
+ifneq (,$(DAPPER_HOST_ARCH))
+
+# Running in Dapper
+
 include Makefile.inc
 
 TARGETS := $(shell ls -p scripts | grep -v -e /)
 
-$(TARGETS): .dapper dapper-image
-	./.dapper -m bind $@
+$(TARGETS):
+	./scripts/$@
+
+.DEFAULT_GOAL := ci
+
+.PHONY: $(TARGETS)
+
+else
 
-# We need the latest image for these targets in shipyard
+# Not running in Dapper
+
+# Shipyard-specific starts
 clusters: dapper-image
 deploy: dapper-image
+# Shipyard-specific ends
 
-.DEFAULT_GOAL := ci
+.dapper:
+	@echo Downloading dapper
+	@curl -sL https://releases.rancher.com/dapper/latest/dapper-`uname -s`-`uname -m` > .dapper.tmp
+	@@chmod +x .dapper.tmp
+	@./.dapper.tmp -v
+	@mv .dapper.tmp .dapper
 
-.PHONY: $(TARGETS)
+%: .dapper dapper-image
+	./.dapper -m bind $(MAKE) $@
+
+shell: .dapper
+	./.dapper -m bind -s
+
+.PHONY: dapper-image shell
```

Comment:
As pointed out by @mkolesnik, this should be pulled out to a separate, included Makefile, to make the overall main Makefile easier to read.